backend/data-store/sql/raw/check_partial_rating_exists.sql

```diff
@@ -1,6 +1,20 @@
--- name: DoesPartialRatingExist :one
+-- name: ValidatePartialRating :one
+
+with get_modified_ts as (
+    select
+        modified at time zone 'utc' as "modified"
+    from project.partial_rating
+    where partial_rating_id = @partial_rating_id
+)
 select (
-    @partial_rating_id::integer in (
-        select partial_rating_id from project.partial_rating
-    )
-)::boolean;
+    case
+        when @partial_rating_id::integer in (
+            select partial_rating_id from project.partial_rating
+        ) then
+            case
+                when @modified::timestamp = (select modified from get_modified_ts) then ''::text
+                else (select to_char(modified, 'YYYY-MM-DD"T"HH24:MI:SS.USZ') from get_modified_ts)::text
+            end
+        else 'id'::text
```

Comment:
While I don't like the convention of returning some specific value (not an error) to indicate that the parameter passed to the query is invalid, I must say that I don't know any other way. Copilot suggests `RAISE` to throw an exception. Maybe this way?

Reply:
Perhaps but we need to 'catch' it on the golang side. If we `RAISE` an error, it will be returned as `error` in golang, which will go this way:
```go
	if err != nil {
		return nil, fmt.Errorf("checking if partial rating exists: %w", err)
	}
```
But this is the same way generic database error would go (connection lost etc.), so in order to intercept that error we would need to do someting like
```go
if err != nil && strings.Contains(err.Error(), "Some custome message we RAISED for race condition") {
	// handle race condition 
} else if err != nil && strings.Contains(err.Error(), "Some custome message we RAISED for missing id") {
	// handle missing id
} else if err != nil {
     // handle err
}
``` 
I'm would not go this way willingly to check if error message contains something -- we don't know beforehand all values that can be there.
I too am not a fan of misusing fields/structs but consider that this string only serves as success indicator. The timestamp returned is mainly for debugging. It's not returned further

Reply:
Alright, let's keep it as it is then.